### ISSUE TITLE
Sky Island Mission wrap up failing due to player index (player 1 vs -1)? May be more missions

### ISSUE THREAD

**Issue by Consoleable** [CleverRaven/Cataclysm-DDA#73557]
### Describe the bug

When completing Sky island missions, getting error page

### Attach save file

[Bonsai-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/15229292/Bonsai-trimmed.tar.gz)

### Steps to reproduce

craft "warped tincture" from nearby desk in attached save

### Expected behavior

no error

### Screenshots

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/107653330/6bafaa9e-8e27-4d56-81df-47bc8002215c)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22631.3447 (23H2)
- Game Version: d6a507f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Translate Complex Dialogue [translate_dialogue],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro],
    Mind Over Matter [mindovermatter],
    Sky Island [skyisland],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex],
    Tamable Wildlife [Tamable_Wildlife]
]

### Additional context

 DEBUG    : mission::wrap_up called, player -1 was assigned, but current player is 1

 FUNCTION : wrap_up
 FILE     : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\mission.cpp
 LINE     : 380
 VERSION  : d6a507f

**Comment by RenechCDDA** [comment]
I'm able to confirm the error is present, but I don't know of any immediate reason for *why*. The fact your player character's ID is 1 instead of -1 is very, very curious.

Did you use the debug menu's export/import functions? Swap characters in some way? Anything you can think of that involved NPCs?

**Comment by Consoleable** [comment]
> I'm able to confirm the error is present, but I don't know of any immediate reason for _why_. The fact your player character's ID is 1 instead of -1 is very, very curious.
> 
> Did you use the debug menu's export/import functions? Swap characters in some way? Anything you can think of that involved NPCs?

no, nothing like that. i used the debug menu to spawn in an item that glitched out, and to change some starting traits(spiritual to gourmand) and that's it

**Comment by RenechCDDA** [comment]
Did you use a character template when originally starting the game?

**Comment by Light-Wave** [comment]
> Did you use a character template when originally starting the game?

I got the same error after having used a Present Character (Last Character) when starting a game. The error came from Scouting 1.
